packages/datadog-plugin-aws-sdk/src/services/lambda.js

```diff
@@ -0,0 +1,56 @@
+'use strict'
+
+const log = require('../../../dd-trace/src/log')
+
+class Lambda {
+  generateTags (params, operation, response) {
+    const tags = {}
+
+    if (!params || !params.FunctionName) return tags
+
+    return Object.assign(tags, {
+      'resource.name': `${operation} ${params.FunctionName}`,
+      'aws.lambda': params.FunctionName
+    })
+  }
+
+  requestInject (span, request, tracer) {
+    const operation = request.operation
+    if (operation === 'invoke') {
+      if (!request.params) {
+        request.params = {}
+      }
+
+      const isSyncInvocation = !request.params.InvocationType ||
+        request.params.InvocationType === 'RequestResponse'
+
+      if (isSyncInvocation) {
+        try {
+          const _datadog = {}
+          tracer.inject(span, 'text_map', _datadog)
+          if (!request.params.ClientContext) {
+            const context = { custom: { _datadog } }
+            request.params.ClientContext = Buffer.from(JSON.stringify(context)).toString('base64')
+          } else {
+            const existingContextJson = Buffer.from(request.params.ClientContext, 'base64').toString('utf-8')
+            const existingContext = JSON.parse(existingContextJson)
+
+            if (existingContext.custom) {
+              existingContext.custom._datadog = _datadog
```

Comment:
Should we always add it to `custom` even when there wasn't an existing `custom` or even `ClientContext`? It would make the logic easier on the receiving side.

Reply:
`custom` will only exist if they explicitly add something. It's an object for everything custom from the client side. I don't follow your statement here.

Reply:
If it doesn't exist can't we just add it? It will simplify the other code paths since then we'd always be checking for `custom`.

Reply:
I think that's what we do in the `else`. But, I might not understand 100% what you say.  If `custom` does not exist we create it here https://github.com/DataDog/dd-trace-js/pull/1166/files/daa552df9853d4134bf4fd8bfa3fd31a9e36533e#diff-c6e77dd2ae520de27ae8300a8d70745178d278a9f62b1b885f6d90a7a3a8c573R41

Reply:
Hmm yeah actually I was referring to when there is no context at all like this: https://github.com/DataDog/dd-trace-js/pull/1166/files/daa552df9853d4134bf4fd8bfa3fd31a9e36533e#diff-c6e77dd2ae520de27ae8300a8d70745178d278a9f62b1b885f6d90a7a3a8c573R31

But it looks like it also adds the `custom` property, I don't know how I missed that.